src/librustdoc/html/render/search_index.rs

```diff
@@ -17,12 +18,25 @@ use crate::html::format::join_with_double_colon;
 use crate::html::markdown::short_markdown_summary;
 use crate::html::render::{self, IndexItem, IndexItemFunctionType, RenderType, RenderTypeId};
 
+/// The serialized search description sharded version
+///
+/// The `index` is a JSON-encoded list of names and other information.
+///
+/// The desc has newlined descriptions, split up by size into 128KiB shards.
+/// For example, `(4, "foo\nbar\nbaz\nquux")`.
+pub(crate) struct SerializedSearchIndex {
+    pub(crate) index: String,
```

Comment:
Is this index actually useful? If you know the index of the item you're looking for (since it's in the `search-index.js` file), you can access it directly into the list, right? Did I misunderstood something?

Reply:
I see, I misunderstood what it was.